### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by ghanzouri**
Implement helper functions to create 3d texture

### CODE REVIEW

**Comment by vettoreldaniele** [comment]
Thanks! This seems like to be a draft PR. I will convert it to draft until it's ready for review

**Comment by ghanzouri** [comment]
It is ready for review, thanks @vettoreldaniele

File: CMakeLists.txt [hunk 1]
```diff
@@ -164,7 +164,7 @@ endif()
 set(CMAKE_CXX_STANDARD 20)
 
 # ------------------------------------------------------------------------------
-# Configure PPX directories.
+# Configure PPX directories. test
```
Comment (vettoreldaniele) [inline CMakeLists.txt:167]: Please remove this test commit

**Comment by ghanzouri** [comment]
How can we add Aria, Jun and Bernhard as reviewers? 
reinert@google.com
ariash@google.com
jyn@google.com

**Comment by vettoreldaniele** [comment]
> How can we add Aria, Jun and Bernhard as reviewers? [reinert@google.com](mailto:reinert@google.com) [ariash@google.com](mailto:ariash@google.com) [jyn@google.com](mailto:jyn@google.com)

I need their GitHub username

File: include/ppx/graphics_util.h [hunk 2]
```diff
@@ -61,6 +68,14 @@ class ImageOptions
         const ImageOptions&          options,
         bool                         useGpu);
 
+    friend Result Create3dImageFromFile(
+    grfx::Queue*                 pQueue,
+    const std::filesystem::path& path,
+    grfx::Image**                ppImage,
+    const uint32_t                  imageDepth,
```
Comment (vettoreldaniele) [inline include/ppx/graphics_util.h:75]: nit: formatting

**Review by vettoreldaniele — COMMENTED** [review]
Can you give a bit more context on what this change enables? I see utilities to load a bitmap into a 3d image, but it's unclear to me what's the intended result. Is the bitmap in a specific `<width, height>` format that gets broken up in layers of the 3d image?

**Comment by ghanzouri** [comment]
Yes exactly, the .png file contains multiple layers of a 3d image. Btw, I did a mistake, I first wanted to create a pull request in my own fork, so that my team can review my changes, should I delete this PR?

**Comment by vettoreldaniele** [comment]
> Yes exactly, the .png file contains multiple layers of a 3d image. Btw, I did a mistake, I first wanted to create a pull request in my own fork, so that my team can review my changes, should I delete this PR?

Sure, you can close this PR and reopen it once you got an OK from your team in your fork. By the way, I would probably add comments or docstring to the function to explain that's what's happening.

**Pull request closed without merging**